lib/graphql/introspection/input_value_type.rb

```diff
@@ -8,8 +8,12 @@
   field :type, !GraphQL::Introspection::TypeType
   field :defaultValue, types.String, "A GraphQL-formatted string representing the default value for this input value." do
     resolve ->(obj, args, ctx) {
-      value = obj.default_value
-      value.nil? ? nil : JSON.generate(obj.type.coerce_result(value), quirks_mode: true)
+      if obj.default_value?
+        value = obj.default_value
+        value.nil? ? 'null' : JSON.generate(obj.type.coerce_result(value), quirks_mode: true)
```

Comment:
I think the answer there is: it depends.

Every type can implement `coerce_result(nil)` the way they want and I didn't want this behaviour to depend on that.

For instance:

```
irb(main):004:0> GraphQL::ListType.new(of_type: GraphQL::ID_TYPE).coerce_result(nil)
=> [nil]
```